### ISSUE TITLE
[BUG] Something went wrong when comparing experiments

### ISSUE THREAD

**Issue by stephanrb3** [mlflow/mlflow#8321]
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

2.2.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Python 3.8 base image from Docker
- **Python version**: 
3.8.16
- **yarn version, if running the dev UI**:
not running from dev ui

### Describe the problem

When clicking compare experiments the UI presents an error: 
Something went wrong
If this error persists, please report an issue [here](https://github.com/mlflow/mlflow/issues).

MLFlow is running is a K8s cluster from an image just using the python version mlflow==2.2.2 on the python3.8 base image. 
We're using Amazon S3 as a backend.
We're using a mysql database from Amazon RDS. 

We're able to log experiments/metrics and get/view them in the UI, however when we click "compare" between to experiment runs we get the something went wrong error.

There are no errors in our mlflow pod logs. However the browser shows there are 502 errors when trying compare. 

### Steps to reproduce the bug

https://user-images.githubusercontent.com/30557024/234381075-39a1ec86-2363-4a9b-8fde-94fc4807f31a.mov

We just compare two experiments and bam. We were not getting this error until we upgraded to mlflow 2.2.2 from 1.x

### Is the console panel in DevTools showing errors relevant to the bug?

<img width="1384" alt="Screenshot 2023-04-25 at 9 30 00 PM" src="https://user-images.githubusercontent.com/30557024/234383331-66f061cb-16f0-4415-9ed8-7abc74c6feff.png">

### Does the network panel in DevTools contain failed requests relevant to the bug?

<img width="1394" alt="Screenshot 2023-04-25 at 9 35 04 PM" src="https://user-images.githubusercontent.com/30557024/234384603-c369fec7-0f7f-4b8f-9bb6-5a5f82cc35b7.png">

**Comment by harupy** [comment]
@stephanrb3 Thanks for reporting this issue. I wasn't able to reproduce  it:

https://user-images.githubusercontent.com/17039389/234437584-1c778141-48e4-4c37-a5b1-a2219cd72728.mov

I don't think this is an MLflow's issue, but an issue in your K8s cluster. Not sure why navigating to the compare page triggers the issue though.

**Comment by stephanrb3** [comment]
Sanity check, it’s the same error still. @harupy
We've upgraded to the official mlflow image and switch to a postgres server. I don't there's a connectivity issue, as we can do literally any other operation without issue.

There's also a similar issue here: https://github.com/mlflow/mlflow/issues/3840 that is almost the same exact problem.

Source line is here:
`const runUuids = JSON.parse(searchValues['?runs']);
`
https://github.com/mlflow/mlflow/blob/v2.1.1/mlflow/server/js/src/experiment-tracking/components/CompareRunPage.js#L57

I tried more debugging on my end.
added a breakpoint and we get an parsing error as soon as we select any run, before even clicking on compare:

```
OTS parsing error: invalid sfntVersion: 1008821359
Failed to decode downloaded font: https://kubeflow.[host]/static-files/static/media/fontawesome-webfont.1e59d2330b4c6deb84b3.ttf
```
This usually occurs trying to load fonts, but getting HTML content instead.
This may or may not be related, not sure.

Finally when comparing three experiments, I was able to find the values it tries to parse during a comparison.
`search: "?runs=%5B%25222aff5917ac904c1b92f6242892bf89eb%2522,%252289914c6a0a7140568299ed4d0d53dd51%2522%5D&experiments=%5B%25222%2522%5D" `
It seems like '%' makes this an invalid json.
